File: csharp/MapleStory.OpenAPI/Src/Dto/Character/CharacterSetEffectInfoDTO.cs
﻿using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 적용 중인 세트 효과 정보
    /// </summary>
    public class CharacterSetEffectInfoDTO
    {
        /// <summary>
        /// 세트 효과 레벨 (장비 수)
        /// </summary>
        [JsonProperty("set_count")]
        public int SetCount { get; set; }

        /// <summary>
        /// 세트 효과
        /// </summary>
        [JsonProperty("set_option")]
        public string SetOption { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterSetEffectInfoDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * Active Set Item effect information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterSetEffectInfoDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterSetEffectInfoDTO {

    /**
     * Set Item effect level (number of equipped items)
     */
    @JsonProperty("set_count")
    private int setCount;

    /**
     * Set Item effect
     */
    @JsonProperty("set_option")
    private String setOption;
}
